### PR TITLE
feat: Added support for Azure Private Link

### PR DESCRIPTION
**Pull request by amanpruthi**
feat:  This Pr creates a Azure private link

### CODE REVIEW

File: modules/app_lb/main.tf [hunk 1]
```diff
@@ -62,10 +62,11 @@ resource "azurerm_application_gateway" "default" {
   }
 
   frontend_ip_configuration {
-    name                          = "${local.frontend_ip_configuration_name}-private"
-    subnet_id                     = var.public_subnet.id
```
Comment (amanpruthi) [inline modules/app_lb/main.tf:66]: Shouldn't this be private ? So that the ip will be picked up from private subnet

**Review by zacharyblasczyk — APPROVED** [review]

File: main.tf [hunk 2]
```diff
@@ -288,15 +332,10 @@ module "wandb" {
...
-        extraEnv = var.weave_wandb_env
       }
 
       mysql = { install = false }
       redis = { install = false }
-
-      parquet = {
-        extraEnv = var.parquet_wandb_env
-      }
     }
```
Comment (zacharyblasczyk) [inline main.tf:339]: Why was this being removed?

File: modules/app_lb/main.tf [hunk 3]
```diff
@@ -108,6 +148,7 @@ resource "azurerm_application_gateway" "default" {
       http_listener,
       backend_http_settings,
       backend_address_pool,
+      private_link_configuration,
```
Comment (zacharyblasczyk) [inline modules/app_lb/main.tf:151]: Why was this being ignored?

File: main.tf (hunk 2 above)
Comment (amanpruthi) [reply main.tf]: Added now, removed by mistake

**Comment by jsbroks** [comment]
This PR is included in version 2.8.0 :tada:

**Pull request merged**